In [1]:
import pandas as pd

- rownames: Row identifier, likely a unique index for each record.
- nr: A unique identifier for each worker.
- year: The year of observation, ranging from 1980 to 1987.
- school: Years of schooling completed by the worker.
- exper: Years of experience, calculated as age - 6 - school, where 6 is a typical starting age for schooling.
- union: Indicates if the worker's wage is set by collective bargaining. Values: yes or no.
- ethn: Ethnicity of the worker. Possible values include black, hisp (Hispanic), or other.
- maried: Indicates marital status. Values: yes or no.
- health: Indicates if the worker has health issues. Values: yes or no.
- wage: The log of hourly wage, a continuous numerical value capturing the natural logarithm of the worker's hourly wage.

In [5]:
Data=pd.read_csv('Wages_and_Education_of_Young_Males_Dataset.csv')
Data

rownames     nr  year  school  exper union   ethn maried health  \
0            1     13  1980      14      1    no  other     no     no   
1            2     13  1981      14      2   yes  other     no     no   
2            3     13  1982      14      3    no  other     no     no   
3            4     13  1983      14      4    no  other     no     no   
4            5     13  1984      14      5    no  other     no     no   
...        ...    ...   ...     ...    ...   ...    ...    ...    ...   
4355      4356  12548  1983       9      8    no  other    yes     no   
4356      4357  12548  1984       9      9   yes  other    yes     no   
4357      4358  12548  1985       9     10    no  other    yes     no   
4358      4359  12548  1986       9     11   yes  other    yes     no   
4359      4360  12548  1987       9     12   yes  other    yes     no   

          wage                          industry  \
0     1.197540       Business_and_Repair_Service   
1     1.853060                  Personal_Service   
2     1.344462       Business_and_Repair_Service   
3     1.433213       Business_and_Repair_Service   
4     1.568125                  Personal_Service   
...        ...                               ...   
4355  1.591879                      Construction   
4356  1.212543                      Construction   
4357  1.765962                      Construction   
4358  1.745894  Professional_and_Related Service   
4359  1.466543             Public_Administration   

                          occupation   residence  
0                    Service_Workers  north_east  
1                    Service_Workers  north_east  
2                    Service_Workers  north_east  
3                    Service_Workers  north_east  
4     Craftsmen, Foremen_and_kindred  north_east  
...                              ...         ...  
4355  Craftsmen, Foremen_and_kindred         NaN  
4356  Craftsmen, Foremen_and_kindred         NaN  
4357  Craftsmen, Foremen_and_kindred         NaN  
4358  Craftsmen, Foremen_and_kindred         NaN  
4359  Craftsmen, Foremen_and_kindred         NaN  

[4360 rows x 13 columns]

In [6]:
Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4360 entries, 0 to 4359
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   rownames    4360 non-null   int64  
 1   nr          4360 non-null   int64  
 2   year        4360 non-null   int64  
 3   school      4360 non-null   int64  
 4   exper       4360 non-null   int64  
 5   union       4360 non-null   object 
 6   ethn        4360 non-null   object 
 7   maried      4360 non-null   object 
 8   health      4360 non-null   object 
 9   wage        4360 non-null   float64
 10  industry    4360 non-null   object 
 11  occupation  4360 non-null   object 
 12  residence   3115 non-null   object 
dtypes: float64(1), int64(5), object(7)
memory usage: 442.9+ KB


In [7]:
Data.dropna(subset='residence',inplace=True)

In [8]:
Data['residence'].unique()

array(['north_east', 'south', 'nothern_central', 'rural_area'],
      dtype=object)

In [11]:
Data.columns

Index(['rownames', 'nr', 'year', 'school', 'exper', 'union', 'ethn', 'maried',
       'health', 'wage', 'industry', 'occupation', 'residence'],
      dtype='object')

In [19]:
Data.set_index('rownames',inplace=True)

# 1: Want to know who are working in insdustry name Business_and_Repair_Service

In [20]:
Under_Bs_and_repair=Data.loc[Data['industry']=='Business_and_Repair_Service'].loc[:]
Under_Bs_and_repair

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
1            13  1980      14      1    no  other     no     no  1.197540   
3            13  1982      14      3    no  other     no     no  1.344462   
4            13  1983      14      4    no  other     no     no  1.433213   
6            13  1985      14      6    no  other     no     no  1.699891   
7            13  1986      14      7    no  other     no     no -0.720263   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4097      11750  1980      10      2    no  other     no     no  1.159187   
4103      11750  1986      10      8    no  other     no     no  1.842593   
4151      11924  1986      12      8    no  other    yes     no  1.416758   
4179      11990  1982      14      4    no  other    yes     no  2.307313   
4181      11990  1984      14      6    no  other    yes     no  2.652110   

                             industry                           occupation  \
rownames                                                                     
1         Business_and_Repair_Service                      Service_Workers   
3         Business_and_Repair_Service                      Service_Workers   
4         Business_and_Repair_Service                      Service_Workers   
6         Business_and_Repair_Service  Managers, Officials_and_Proprietors   
7         Business_and_Repair_Service  Managers, Officials_and_Proprietors   
...                               ...                                  ...   
4097      Business_and_Repair_Service               Operatives_and_kindred   
4103      Business_and_Repair_Service       Craftsmen, Foremen_and_kindred   
4151      Business_and_Repair_Service                 Clerical_and_kindred   
4179      Business_and_Repair_Service                        Sales_Workers   
4181      Business_and_Repair_Service                        Sales_Workers   

           residence  
rownames              
1         north_east  
3         north_east  
4         north_east  
6         north_east  
7         north_east  
...              ...  
4097           south  
4103      north_east  
4151           south  
4179      north_east  
4181      north_east  

[237 rows x 12 columns]

In [16]:
Under_Bs_and_repair['industry'].value_counts()

industry
Business_and_Repair_Service    237
Name: count, dtype: int64

- TAKEAWAY: There were 237 males working under business and repair service 

# 2: I want to know how many married and unmarried people in business and repair service

In [17]:
Under_Bs_and_repair['maried'].value_counts()

maried
no     147
yes     90
Name: count, dtype: int64

- TAKEAWAY : There were 90 married workers and 147 unmarried workers majority of them are umarried

# 3: I want to know which one of the nr having more experience in business and repair service

In [30]:
Under_Bs_and_repair[['exper','nr']].sort_values(by='exper',ascending=False)

exper    nr
rownames             
2056         16  4261
3719         13  9603
1976         13  4046
1975         12  4046
3718         12  9603
...         ...   ...
2305          2  4881
1090          2  2329
1337          2  2916
3377          2  8656
1             1    13

[237 rows x 2 columns]

- TAKEAWAY: Worker NO:4261 having most experience of 16 Years

# 4: Want to know the workers whose residence is nothern_central and there years of compliting school

In [35]:
north_central=Data[Data['residence']=='nothern_central'].loc[:,['school','nr','residence']]
north_central

school     nr        residence
rownames                                
80            10    166  nothern_central
201           12    483  nothern_central
202           12    483  nothern_central
203           12    483  nothern_central
204           12    483  nothern_central
...          ...    ...              ...
4225          12  12088  nothern_central
4226          12  12088  nothern_central
4227          12  12088  nothern_central
4228          12  12088  nothern_central
4266          11  12221  nothern_central

[964 rows x 3 columns]

In [36]:
north_central['school'].unique()

array([10, 12, 14, 11, 13, 15,  8, 16,  6,  9], dtype=int64)

- TAKEAWAY:These are the result of residence nothern_central and there school years

# 5: I want to get the worker ID whose under the industry:Business_and_Repair_Service and occupation is Operatives_and_kindred 

In [39]:
BS_OP=Data[(Data['occupation']=='Operatives_and_kindred') & (Data['industry']=='Business_and_Repair_Service')].loc[:,['nr','industry','occupation']]
BS_OP

nr                     industry              occupation
rownames                                                            
29           45  Business_and_Repair_Service  Operatives_and_kindred
57          150  Business_and_Repair_Service  Operatives_and_kindred
58          150  Business_and_Repair_Service  Operatives_and_kindred
59          150  Business_and_Repair_Service  Operatives_and_kindred
60          150  Business_and_Repair_Service  Operatives_and_kindred
63          150  Business_and_Repair_Service  Operatives_and_kindred
145         309  Business_and_Repair_Service  Operatives_and_kindred
146         309  Business_and_Repair_Service  Operatives_and_kindred
147         309  Business_and_Repair_Service  Operatives_and_kindred
149         309  Business_and_Repair_Service  Operatives_and_kindred
151         309  Business_and_Repair_Service  Operatives_and_kindred
462         947  Business_and_Repair_Service  Operatives_and_kindred
874        1930  Business_and_Repair_Service  Operatives_and_kindred
1337       2916  Business_and_Repair_Service  Operatives_and_kindred
1579       3282  Business_and_Repair_Service  Operatives_and_kindred
1623       3353  Business_and_Repair_Service  Operatives_and_kindred
1624       3353  Business_and_Repair_Service  Operatives_and_kindred
1725       3503  Business_and_Repair_Service  Operatives_and_kindred
2449       5252  Business_and_Repair_Service  Operatives_and_kindred
2624       5665  Business_and_Repair_Service  Operatives_and_kindred
2934       7087  Business_and_Repair_Service  Operatives_and_kindred
3233       8364  Business_and_Repair_Service  Operatives_and_kindred
3275       8415  Business_and_Repair_Service  Operatives_and_kindred
3306       8520  Business_and_Repair_Service  Operatives_and_kindred
4097      11750  Business_and_Repair_Service  Operatives_and_kindred

In [40]:
len(BS_OP)

25

- TAKEAWAY: There were 25 workers who is under the industry Business_and_Repair_Service and occupation:Operative and kindred

# 6: Top 5 industry depends on wages porvided to the workers 

In [46]:
top_5=Data.sort_values(by='wage',ascending=False).head(5)
top_5

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                   
3054      7784  1985      11      6    no  other     no     no  4.051860   
3053      7784  1984      11      5    no  other     no     no  3.777492   
3051      7784  1982      11      3    no  other     no     no  3.472707   
1734      3525  1985      14      6    no  other     no     no  3.449363   
1327      2868  1986      13      9    no  other    yes     no  3.397526   

                                  industry  \
rownames                                     
3054                               Finance   
3053                               Finance   
3051           Business_and_Repair_Service   
1734      Professional_and_Related Service   
1327                                 Trade   

                                   occupation   residence  
rownames                                                   
3054                            Sales_Workers       south  
3053                            Sales_Workers       south  
3051                            Sales_Workers       south  
1734                            Sales_Workers       south  
1327      Managers, Officials_and_Proprietors  north_east

In [47]:
top_5.loc[:,['wage','industry']]

wage                          industry
rownames                                            
3054      4.051860                           Finance
3053      3.777492                           Finance
3051      3.472707       Business_and_Repair_Service
1734      3.449363  Professional_and_Related Service
1327      3.397526                             Trade

- TAKEAWAY These are the top 5 industry 

# 7: The worker details whose ethinicity black and hisp 

In [50]:
Data['ethn'].unique()

array(['other', 'black', 'hisp'], dtype=object)

In [51]:
Data[(Data['ethn']=='black')|(Data['ethn']=='hisp')].loc[:]

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
169         383  1980      12      2    no  black     no     no  1.462888   
170         383  1981      12      3    no  black     no     no  1.570217   
171         383  1982      12      4   yes  black     no     no  1.656602   
172         383  1983      12      5    no  black     no     no  1.725410   
173         383  1984      12      6    no  black     no     no  1.772159   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4260      12220  1983      12      5   yes   hisp    yes     no  2.064561   
4261      12220  1984      12      6   yes   hisp    yes     no  1.954480   
4262      12220  1985      12      7   yes   hisp    yes     no  2.011084   
4263      12220  1986      12      8   yes   hisp    yes     no  2.074398   
4264      12220  1987      12      9   yes   hisp    yes     no  2.217754   

                industry                      occupation   residence  
rownames                                                              
169        Manufacturing          Operatives_and_kindred  north_east  
170        Manufacturing          Operatives_and_kindred  north_east  
171        Manufacturing          Operatives_and_kindred  north_east  
172        Manufacturing            Clerical_and_kindred  north_east  
173        Manufacturing            Clerical_and_kindred  north_east  
...                  ...                             ...         ...  
4260      Transportation  Craftsmen, Foremen_and_kindred       south  
4261      Transportation  Craftsmen, Foremen_and_kindred       south  
4262      Transportation  Craftsmen, Foremen_and_kindred       south  
4263      Transportation  Craftsmen, Foremen_and_kindred       south  
4264      Transportation  Craftsmen, Foremen_and_kindred       south  

[739 rows x 12 columns]

- TAKEAWAY: There are 739 workers whose ethinicity is black and hisp 

# 8: I want to know the workers details whose ethinicity is black and the residence is south

In [53]:
Data['residence'].unique()

array(['north_east', 'south', 'nothern_central', 'rural_area'],
      dtype=object)

In [56]:
Data[(Data['ethn']=='black')&(Data['residence']=='south')]

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
585        1156  1980      12      5    no  black     no     no  1.302818   
586        1156  1981      12      6    no  black     no     no  1.330725   
587        1156  1982      12      7    no  black     no     no  1.299566   
588        1156  1983      12      8    no  black     no     no  1.631099   
589        1156  1984      12      9    no  black     no     no  1.659154   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4076      10666  1983      11      6    no  black    yes     no  0.451336   
4077      10666  1984      11      7    no  black    yes     no  0.802813   
4078      10666  1985      11      8    no  black    yes     no  0.706570   
4079      10666  1986      11      9    no  black    yes     no  1.074726   
4080      10666  1987      11     10    no  black     no     no  0.694628   

                                  industry              occupation residence  
rownames                                                                      
585                                  Trade         Service_Workers     south  
586                                  Trade         Service_Workers     south  
587                                  Trade         Service_Workers     south  
588                                  Trade         Service_Workers     south  
589                                  Trade         Service_Workers     south  
...                                    ...                     ...       ...  
4076                                 Trade  Operatives_and_kindred     south  
4077           Business_and_Repair_Service    Laborers_and_farmers     south  
4078      Professional_and_Related Service         Service_Workers     south  
4079                                 Trade    Clerical_and_kindred     south  
4080                                 Trade    Clerical_and_kindred     south  

[225 rows x 12 columns]

- TAKEAWAY:225 Workers whose ethinicity is black and residency is south 

# 9: I want to get the details of the year 1983 

In [59]:
In_1983=Data[Data['year']==1983].loc[:]
In_1983

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
4            13  1983      14      4    no  other     no     no  1.433213   
12           17  1983      13      7    no  other     no     no  1.725410   
28           45  1983      12      5    no  other     no     no  1.740914   
36          110  1983      12      8    no  other     no     no  2.202516   
44          120  1983      10      5    no  other     no     no  1.320287   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4260      12220  1983      12      5   yes   hisp    yes     no  2.064561   
4276      12245  1983      14      4    no  other     no     no  1.181936   
4316      12433  1983      15      5    no  other     no     no  1.268716   
4324      12451  1983      14      5    no  other    yes     no  2.016517   
4332      12477  1983      12      7   yes  other    yes     no  2.171697   

                                  industry  \
rownames                                     
4              Business_and_Repair_Service   
12                                   Trade   
28                           Manufacturing   
36        Professional_and_Related Service   
44                                   Trade   
...                                    ...   
4260                        Transportation   
4276      Professional_and_Related Service   
4316      Professional_and_Related Service   
4324      Professional_and_Related Service   
4332                         Manufacturing   

                                   occupation   residence  
rownames                                                   
4                             Service_Workers  north_east  
12        Managers, Officials_and_Proprietors  north_east  
28                     Operatives_and_kindred  north_east  
36        Managers, Officials_and_Proprietors  north_east  
44                            Service_Workers  north_east  
...                                       ...         ...  
4260           Craftsmen, Foremen_and_kindred       south  
4276                     Clerical_and_kindred       south  
4316                     Clerical_and_kindred       south  
4324      Professional, Technical_and_kindred       south  
4332           Craftsmen, Foremen_and_kindred  north_east  

[391 rows x 12 columns]

- TAKEAWAY: these are the result of the year 1983  

# 10: In 1983 which industry recruted the workers most

In [60]:
In_1983['industry'].value_counts()

industry
Trade                               108
Manufacturing                       106
Professional_and_Related Service     35
Business_and_Repair_Service          32
Transportation                       31
Public_Administration                17
Construction                         17
Finance                              14
Agricultural                         14
Personal_Service                      7
Entertainment                         5
Mining                                5
Name: count, dtype: int64

- TAKEAWAY : In 1983 Trade industry having 108 data details 

# 11: I want to get the total wages of whole data (from 1980 to 1987)

In [62]:
Data.pivot_table(index='year',values='wage',aggfunc=sum,margins=True,margins_name='Total_Wage')

wage
year                   
1980         562.470304
1981         606.021489
1982         614.157300
1983         637.838571
1984         651.068419
1985         671.342576
1986         698.630033
1987         700.495929
Total_Wage  5142.024621

- TAKEAWAY : These are the total wages of all years and total of all years

# 12: I want to get the deatils of the worker_id 13

In [65]:
data_13=Data[Data['nr']==13].loc[:]
data_13

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                 
1         13  1980      14      1    no  other     no     no  1.197540   
2         13  1981      14      2   yes  other     no     no  1.853060   
3         13  1982      14      3    no  other     no     no  1.344462   
4         13  1983      14      4    no  other     no     no  1.433213   
5         13  1984      14      5    no  other     no     no  1.568125   
6         13  1985      14      6    no  other     no     no  1.699891   
7         13  1986      14      7    no  other     no     no -0.720263   
8         13  1987      14      8    no  other     no     no  1.669188   

                             industry                           occupation  \
rownames                                                                     
1         Business_and_Repair_Service                      Service_Workers   
2                    Personal_Service                      Service_Workers   
3         Business_and_Repair_Service                      Service_Workers   
4         Business_and_Repair_Service                      Service_Workers   
5                    Personal_Service       Craftsmen, Foremen_and_kindred   
6         Business_and_Repair_Service  Managers, Officials_and_Proprietors   
7         Business_and_Repair_Service  Managers, Officials_and_Proprietors   
8         Business_and_Repair_Service  Managers, Officials_and_Proprietors   

           residence  
rownames              
1         north_east  
2         north_east  
3         north_east  
4         north_east  
5         north_east  
6         north_east  
7         north_east  
8         north_east

# 13: I want to know worker_id 13 ,which of the following industry he works

In [66]:
data_13['industry'].unique()

array(['Business_and_Repair_Service', 'Personal_Service'], dtype=object)

 - TAKEWAY: He works in two industries as follow: 'Business and repair service' and 'Personal service'

# 14: Get the details of workers having experience more than 5  and school years eqaul to or more than 15

In [71]:
Data[Data['exper']==5].loc[:]

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
5            13  1984      14      5    no  other     no     no  1.568125   
10           17  1981      13      5    no  other     no     no  1.518398   
28           45  1983      12      5    no  other     no     no  1.740914   
33          110  1980      12      5   yes  other     no     no  1.948775   
44          120  1983      10      5    no  other     no     no  1.320287   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4260      12220  1983      12      5   yes   hisp    yes     no  2.064561   
4277      12245  1984      14      5    no  other     no     no  1.148810   
4316      12433  1983      15      5    no  other     no     no  1.268716   
4324      12451  1983      14      5    no  other    yes     no  2.016517   
4330      12477  1981      12      5   yes  other     no     no  2.222542   

                                  industry  \
rownames                                     
5                         Personal_Service   
10                                   Trade   
28                           Manufacturing   
33        Professional_and_Related Service   
44                                   Trade   
...                                    ...   
4260                        Transportation   
4277                         Entertainment   
4316      Professional_and_Related Service   
4324      Professional_and_Related Service   
4330                         Manufacturing   

                                   occupation   residence  
rownames                                                   
5              Craftsmen, Foremen_and_kindred  north_east  
10        Managers, Officials_and_Proprietors  north_east  
28                     Operatives_and_kindred  north_east  
33                       Clerical_and_kindred  north_east  
44                            Service_Workers  north_east  
...                                       ...         ...  
4260           Craftsmen, Foremen_and_kindred       south  
4277      Managers, Officials_and_Proprietors       south  
4316                     Clerical_and_kindred       south  
4324      Professional, Technical_and_kindred       south  
4330           Craftsmen, Foremen_and_kindred  north_east  

[377 rows x 12 columns]

# 15:I want to get the details of the workers who is a union worker in North East

In [75]:
union=Data[(Data['union']=='yes')&(Data['residence']=='north_east')].loc[:]
union

nr  year  school  exper union   ethn maried health      wage  \
rownames                                                                    
2            13  1981      14      2   yes  other     no     no  1.853060   
25           45  1980      12      2   yes  other     no     no  1.894115   
26           45  1981      12      3   yes  other     no     no  1.471159   
33          110  1980      12      5   yes  other     no     no  1.948775   
64          150  1987      12      9   yes  other     no     no  0.831460   
...         ...   ...     ...    ...   ...    ...    ...    ...       ...   
4332      12477  1983      12      7   yes  other    yes     no  2.171697   
4333      12477  1984      12      8   yes  other    yes     no  2.168054   
4334      12477  1985      12      9   yes  other    yes     no  2.133687   
4335      12477  1986      12     10   yes  other    yes     no  2.113619   
4336      12477  1987      12     11   yes  other    yes     no  2.202486   

                                  industry                      occupation  \
rownames                                                                     
2                         Personal_Service                 Service_Workers   
25                          Transportation            Laborers_and_farmers   
26                          Transportation          Operatives_and_kindred   
33        Professional_and_Related Service            Clerical_and_kindred   
64             Business_and_Repair_Service                 Service_Workers   
...                                    ...                             ...   
4332                         Manufacturing  Craftsmen, Foremen_and_kindred   
4333                         Manufacturing          Operatives_and_kindred   
4334                         Manufacturing  Craftsmen, Foremen_and_kindred   
4335                         Manufacturing          Operatives_and_kindred   
4336                         Manufacturing                 Service_Workers   

           residence  
rownames              
2         north_east  
25        north_east  
26        north_east  
33        north_east  
64        north_east  
...              ...  
4332      north_east  
4333      north_east  
4334      north_east  
4335      north_east  
4336      north_east  

[189 rows x 12 columns]

- TAKEAWAY: There are 189 workers 

# 16: I want to get the workers_id whose wage is more than 1 and they having no health issues

In [77]:
Data[(Data['wage']>1)&(Data['health']=='no')].loc[:,['nr','wage','health']]

nr      wage health
rownames                        
1            13  1.197540     no
2            13  1.853060     no
3            13  1.344462     no
4            13  1.433213     no
5            13  1.568125     no
...         ...       ...    ...
4332      12477  2.171697     no
4333      12477  2.168054     no
4334      12477  2.133687     no
4335      12477  2.113619     no
4336      12477  2.202486     no

[2812 rows x 3 columns]